syz-manager/manager.go

```diff
@@ -1007,16 +1016,28 @@ func (mgr *Manager) saveRepro(res *repro.Result, stats *repro.Stats, hub bool) {
 		// 2. Repro re-tried 3 times and still got corrupted report at the end,
 		//    so maybe corrupted report detection is broken.
 		// 3. Reproduction is expensive so it's good to persist the result.
+
+		report := repro.Report
+		output := report.Output
+		if res.strace != nil && res.strace.Report != nil &&
+			res.strace.Report.Title == repro.Report.Title {
+			// If syzkaller managed to successfully run the repro with strace, send
+			// the report and the output generated under strace.
+			// We require the title to be the same because otherwise syzbot might not
```

Comment:
I don't understand this comment and logic.

Reply:
Syzbot triggered a bug with the title `A`.
pkg/repro returned a repro with the title `B`.
We ran that under strace and got a crash with the title `C`.

Right now we save `A` and `B` as if they were separate crashes, which is reasonable. If someone sends a patch for bug `B`, syzbot can run the repro and see if the same bug is fixed or not. 

If we save `C` also a separate bug, we should ideally also teach syzbot to always reproduce this bug under strace, as it may have affected the kernel behavior (and let do B != C). Which does not really seem to be worth it, easier to just discard strace output in such case and pretend we never tried it.

Reply:
Ack.